offline/coturn.md

```diff
@@ -0,0 +1,471 @@
+
+``` 
+
+This section, with the secrets, are what we want to copy into our `secret.yaml` file for Coturn.
+
+Create/edit a `secret.yaml` file inside the `values/coturn` folder:
+
+```bash
+
+nano values/coturn/secret.yaml
+
+```
+
+Add the following configuration to the `secret.yaml` file:
+
+```yaml
+
+# Path is .secrets.
+secrets:
+  zrestSecrets:
+    - "Ob4C52U8WPwv[...]QUy724p1n"
+
+``` 
+
+Here, the value for `secrets.zrestSecrets` is the same as `brig.secrets.turn.secret` from the wire-server secrets.
+
+## Configure the Coturn labels to select on which machine(s) it will run.
+
+Next, we must select on which machine Coturn will run.
+
+In this example, we've decided it will run on the third kubernetes node, `kubenode3`, which has an IP address of `192.168.122.23`.
+
+We've set the `nodeSelector` in the `values.yaml` file to select the `coturn` role, this and machine we label with the `wire.com/role: coturn` label will be selected to run Coturn.
+
+So we need to label the `kubenode3` machine with the `wire.com/role: coturn` label.
+
+We do this by running:
+
+```bash
+
+d kubectl label node kubenode3 wire.com/role=coturn
+
+```
+
+By default, only one machine will be selected to run Coturn. 
+
+If you want to run Coturn on multiple machines, you must:
+
+1. Add the `wire.com/role: coturn` label to multiple machines.
+
+2. Change the `replicaCount` in the `charts/coturn/values.yaml` file to the number of machines you want to run Coturn on.
+
+## Configure the SFT labels for Coturn and SFT to share a port range. 
+
+```{note}
+
+Note this step is also documented in the Switch to [ Wire install docs](docs_ubuntu_22.04.md)
+
+``` 
+
+We must make sure that Coturn pods and SFT pods do not run on the same kubernetes nodes.
+
+This means we must label the kubernetes nodes to run on nodes that we did not select to run Coturn in the previous step.
+
+In this example, we've decided to run Coturn on the first kubernetes node, `kubenode1`, which has an IP address of `192.168.122.21`.
+
+First we make sure the SFT chart is configured to only run on kubernetes nodes with the right label (`sftd`).
+
+Edit the `values/sftd/values.yaml` file:
+
+```bash
+
+nodeSelector:
+  wire.com/role: sftd
+
+``` 
+
+Then we label the `kubenode1` machine with the `wire.com/role: coturn` label:
+
+```bash
+
+d kubectl label node kubenode1 wire.com/role=sftd
+
+```
+
+If we want to run SFT on multiple nodes, the procedure is the same as the one documented above for running Coturn on multiple nodes.
+
+We now should have Coturn configured to run on one or more kubernetes node(s), and SFT configured to run on one or more kubernetes node(s), and the two should not run on the same node(s)/overlap.
+
+Before moving on, we must also re-deploy SFT's chart to apply the new configuration:
+
+```bash
+
+d helm upgrade --install sftd ./charts/sftd --set 'nodeSelector.wire\.com/role=sftd' --values values/sftd/values.yaml
+
+```
+
+## Configure the port redirection in Nftables.
```

Comment:
we should mark this section as only if you are running behind a nftables managed firewall.

Reply:
fixed in latest commit:

```{note}

Note: This section is only relevant if you are running Wire-Server/Coturn/SFT behind a `nftables`-managed firewall.

```